ansible_builder/main.py

```diff
@@ -329,26 +329,46 @@ def prepare_galaxy_install_steps(self):
             self.steps.extend(GalaxyInstallSteps(CONTEXT_FILES['galaxy']))
         return self.steps
 
-    def prepare_pip_steps(self, pip_lines):
-        if ''.join(pip_lines).strip():  # only use file if it is non-blank
-            pip_file = os.path.join(self.build_context, PIP_COMBINED)
-            write_file(pip_file, pip_lines)
-            self.steps.extend(PipSteps(PIP_COMBINED))
+    def prepare_assemble_steps(self):
+        requirements_file_exists = os.path.exists(os.path.join(self.build_context, PIP_COMBINED))
+        if requirements_file_exists:
+            self.steps.append("ADD {0} /tmp/src/requirements.txt".format(PIP_COMBINED))
+
+        bindep_exists = os.path.exists(os.path.join(self.build_context, BINDEP_COMBINED))
+        if bindep_exists:
+            self.steps.append("ADD {0} /tmp/src/bindep.txt".format(BINDEP_COMBINED))
+
+        if requirements_file_exists or bindep_exists:
+            self.steps.append("RUN assemble")
 
         return self.steps
 
-    def prepare_system_steps(self, bindep_output):
-        if ''.join(bindep_output).strip():
-            system_file = os.path.join(self.build_context, BINDEP_OUTPUT)
-            write_file(system_file, bindep_output)
-            self.steps.extend(BindepSteps(BINDEP_OUTPUT))
+    def prepare_system_runtime_deps_steps(self):
+        requirements_file_exists = os.path.exists(os.path.join(self.build_context, PIP_COMBINED))
+        bindep_exists = os.path.exists(os.path.join(self.build_context, BINDEP_COMBINED))
+
+        if requirements_file_exists or bindep_exists:
+            self.steps.extend([
+                "COPY --from=builder /output/ /output/",
+                "RUN /output/install-from-bindep",
+                "RUN rm -rf /output"
```

Comment:
since it's another layer, I don't know that you save any size by removing the output folder.

Reply:
Let me run some tests and report back.

Reply:
AFAIK, until podman supports buildkit, the COPY command will put the wheels on disk.

I've experimented about standing up an HTTP server in the python-builder layer, they using pip install and set index-url to localhost, but still WIP.

At worst case, it might add a few MBs right now

Reply:
I expected this folder is requirements files, so yeah the disk size should be totally trivial. It's more that deleting it isn't truly hiding it, if something is in there that shouldn't be.

Reply:
@AlanCoding this directory contains the pre-built wheels that we are installing in the final image. I like your idea of leaving this metadata behind, so I'm going to switch to only removing `/output/wheels`